File: .NET/Microsoft.Recognizers.Text/Matcher/AaNode.cs
using System.Collections.Generic;
using System.Linq;

namespace Microsoft.Recognizers.Text.Matcher
{
    public class AaNode<T> : Node<T>
    {
        public AaNode()
        {

        }

        public AaNode(T c, int depth)
        {
            Word = c;
            Depth = depth;
        }

        public AaNode(T c, int depth, AaNode<T> parent)
        {
            Word = c;
            Depth = depth;
            Parent = parent;
        }

        public T Word { get; set; }
        public int Depth { get; set; }
        public AaNode<T> Parent { get; set; }

        public AaNode<T> Fail { get; set; } = null;

        public new AaNode<T> this[T c]
        {
            get { return Children != null && Children.ContainsKey(c) ? Children[c] as AaNode<T> : null; }
            set
            {
                if (Children == null)
                {
                    Children = new Dictionary<T, Node<T>>();
                }

                Children[c] = value;
            }
        }

        public new IEnumerator<AaNode<T>> GetEnumerator()
        {
            return Children?.Values.Select(child => child as AaNode<T>).GetEnumerator();
        }

        public override string ToString()
        {
            return Word.ToString();
        }
    }
}

File: Python/libraries/recognizers-text/recognizers_text/matcher2/aa_node.py
from .node import Node


class AaNode(Node):

    def __init__(self, c: [] = [], depth: int = 0, parent=[]):
        self.__word = c,
        self.__depth = depth,
        self.__parent = parent,
        self.__fail = 0

    def __getitem__(self, item):
        if self.children is not None and item in self.children:
            return self.children[item]
        else:
            return None

    def __setitem__(self, key, value):
        if self.children is None:
            self.children = {}

        self.children[key] = value

    @property
    def word(self) -> []:
        return self.__word

    @word.setter
    def word(self, word):
        self.__word = word

    @property
    def depth(self):
        return self.__depth

    @depth.setter
    def depth(self, depth):
        self.__depth = depth

    @property
    def parent(self):
        return self.__parent

    @parent.setter
    def parent(self, parent):
        self.__parent = parent

    @property
    def fail(self):
        return self.__fail

    @fail.setter
    def fail(self, fail):
        self.__fail = fail

    def to_string(self):
        return str(self.__word)
